File: nest-sso-backend/src/user/user.controller.ts
import { Controller, Get, UseGuards } from '@nestjs/common';
import { User } from './user.interface';
import {
  AuthGuard,
  ResourceGuard,
  Roles,
  AuthenticatedUser,
  RoleGuard,
} from 'nest-keycloak-connect';

@Controller('users')
export class UserController {
  constructor() {}

  @Get('me')
  @UseGuards(AuthGuard)
  getProfile(@AuthenticatedUser() user: User) {
    return user;
  }

  @UseGuards(AuthGuard, ResourceGuard, RoleGuard)
  @Roles({ roles: ['realm:admin'] })
  @Get('admin-data')
  getAdminData() {
    return { msg: 'Solo para admin' };
  }
}

File: spring-sso-backend/src/main/java/com/ejemplo/backend/controller/UserController.java
package com.ejemplo.backend.controller;

import org.springframework.security.access.prepost.PreAuthorize;
import org.springframework.security.oauth2.server.resource.authentication.JwtAuthenticationToken;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.Map;


@RestController
@RequestMapping("/users")
public class UserController {

    @GetMapping("/me")
    public Object getProfile(JwtAuthenticationToken token) {
        return token.getToken().getClaims();
    }

    @PreAuthorize("hasAuthority('ROLE_realm_admin')")
    @GetMapping("/admin-data")
    public Object getAdminData() {
        return Map.of("msg", "Solo para admin");
    }


}
